### DISCUSSION TITLE
Is it possible to "assign" multiple times within a single "action"?

### DISCUSSION THREAD

**Discussion by Sheldonfrith** [statelyai/xstate#1775] [Q&A] [answered]
I'm trying to use a single "action" to spawn multiple actors. However as each "spawn" has to be within an "assign" call I'm a little stuck. How can I queue multiple "assign" calls within a single "actor"? I cannot find any examples of this in the docs.
Thanks!

**Comment by davidkpiano** [comment, marked as answer]
You can use the [`pure()` action creator](https://xstate.js.org/docs/guides/actions.html#pure-action) to do multiple assigns, but it sounds like you might not need to:

```js
actions: assign({
  someActors: () => {
    return [/* array of things */].map(thing => spawn(createSomeMachine(thing)))
  }
});
```

**Reply by Sheldonfrith** [reply]
Thanks!
